plugins/catalog-backend/src/database/CommonDatabase.ts

```diff
@@ -235,17 +238,43 @@ export class CommonDatabase implements Database {
                 }
               }
             });
-
           this.andWhere('id', 'in', matchQuery);
         }
       });
     }
 
-    const rows = await entitiesQuery
+    entitiesQuery = entitiesQuery
       .select('entities.*')
       .orderBy('full_name', 'asc');
 
-    return this.toEntityResponses(tx, rows);
+    const { limit, offset } = parsePagination(request?.pagination);
+    if (limit !== undefined) {
```

Comment:
Wouldn't this be sufficient?
```suggestion
    if (limit) {
```